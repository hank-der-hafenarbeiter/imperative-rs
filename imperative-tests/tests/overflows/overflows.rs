use imperative_rs::InstructionSet;


#[derive(InstructionSet)]
enum Instructionset {
    #[ opcode = "0x0x_xx" ]
    A{x:u8},
    #[ opcode = "0x1x_xx_xx" ]
    B{x:u16},
    #[ opcode = "0x2x_xx_xx_xx_xx" ]
    C{x:u32},
    #[ opcode = "0x3x_xx_xx_xx_xx_xx_xx_xx_xx" ]
    D{x:u64},
}

fn main() {
    let mut mem = [0xffu8; 10];
    //0x3x_xx_xx_xx_xx_xx_xx_xx_xx
    mem[0] = 0x3f;
    let _ = Instructionset::decode(&mem);
}

