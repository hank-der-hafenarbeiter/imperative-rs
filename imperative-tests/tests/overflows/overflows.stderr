error: Variable x (with symbol: x) has fewer bits (8) than positions in opcode (12). (e.g. u8 with 9 bits in opcode)
 --> $DIR/overflows.rs:7:7
  |
7 |     A{x:u8},
  |       ^

error[E0599]: no variant or associated item named `decode` found for enum `Instructionset` in the current scope
  --> $DIR/overflows.rs:20:29
   |
5  | enum Instructionset {
   | ------------------- variant or associated item `decode` not found here
...
20 |     let _ = Instructionset::decode(&mem);
   |                             ^^^^^^ variant or associated item not found in `Instructionset`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `decode`, perhaps you need to implement it:
           candidate #1: `InstructionSet`
